modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/resources/manifest/ExportedArtifactsCollector.java

```diff
@@ -151,13 +153,18 @@ private void collectExportedClass(MetadataType type) {
 
       @Override
       public void visitDictionary(DictionaryType dictionaryType) {
-        dictionaryType.getKeyType().accept(this);
-        dictionaryType.getValueType().accept(this);
+        addType(dictionaryType.getKeyType(), exportedClasses);
+        addType(dictionaryType.getValueType(), exportedClasses);
       }
 
       @Override
       public void visitArrayType(ArrayType arrayType) {
-        arrayType.getType().accept(this);
+        addType(arrayType.getType(), exportedClasses);
```

Comment:
what is the root cause of the issue? why does this fix it? This solution is not valid since this wouldn't catch the case of List<Map<String, Apple>> . The reason why the original code used the visitor with the array's type was to allow for those combinations.

Why does this fix the issue? Isn't the real problem in `visitObjectField`?